packages/autocomplete-tools/@types/fig/index.d.ts

```diff
@@ -3,25 +3,106 @@ export {};
 
 declare global {
   export namespace Fig {
-    // All the available templates
     type TemplateStrings = "filepaths" | "folders" | "history";
 
+    /**
+     * A template which is a single TemplateString or an array of TemplateStrings.
+     *
+     * @remark
+     * Templates are generators prebuilt by Fig. Here are the three templates:
+     * * filepaths: show folders and filepaths. Allow autoexecute on filepaths
+     * * folders: show folders only. Allow autoexecute on folders
+     * * history: show suggestions for all items in history matching this pattern
+     *
+     * @example
+     * `cd` uses the "folders" template
+     * `ls` useed  ["filepaths", "folders"]. Why both? Because if I `ls` a directory, we want to enable a user to autoexecute on this directory. If we just did "filepaths" they couldn't autoexecute.
```

Comment:
Typo
```suggestion
     * `ls` used  ["filepaths", "folders"]. Why both? Because if I `ls` a directory, we want to enable a user to autoexecute on this directory. If we just did "filepaths" they couldn't autoexecute.
```